simple_ado/http_client.py

```diff
@@ -178,8 +178,8 @@ def _track_rate_limit(self, response: requests.Response) -> None:
 
     @retry(
         retry=(
-            retry_if_exception(_is_connection_failure)
-            | retry_if_exception(_is_retryable_get_failure)
+            retry_if_exception(_is_connection_failure)  # type: ignore
```

Comment:
Ignoring type checks here as mypy is complaining about "Argument 1 to "retry_if_exception" has incompatible type "Callable[[Exception], bool]"; expected "Callable[[BaseException], bool]"", which seems to be a false alarm.